### PR TITLE
Assignment_4

### PR DESCRIPTION
**Pull request by meta-sabir-mohammad**
18/07/2019

### CODE REVIEW

File: ArrOperation.java [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+ *
+ */
+public class ArrOperation {
+
+	/**
+	 * Find size of largest mirror section in positive integer array
+	 * @param inputArray an array of positive integer
+	 * @return size of largest mirror section in the array
+	 * @throws AssertionError when array is empty 
+	 * @throws AssertionError when array contain negative number
```
Comment (ghost) [inline ArrOperation.java:18]: Use single "@throws AssertionError" and add both condition detail. Use different throw statement if different type of exception.

File: ArrOperation.java [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+	 */
+	public int getSizeOfLargestMirrorSectionInArray(int inputArray[]) throws AssertionError{
+		
+		//input validation
+		if(inputArray.length == 0){
+			throw new AssertionError("Null Array");
+		}else{
+			boolean validInput = inputValidation(inputArray);
+			if(validInput == false){
```
Comment (ghost) [inline ArrOperation.java:27]: Use boolean values directly like if(validInput) or if(!validInput).

Comment (meta-sabir-mohammad) [reply ArrOperation.java:27]: ok sir

File: ArrOperation.java (hunk 1 above)
Comment (meta-sabir-mohammad) [reply ArrOperation.java:18]: ok sir

**Pull request closed without merging**